coriolis_openstack_utils/instances.py

```diff
@@ -120,3 +123,99 @@ def validate_migration_options(
+                         (set(instances_names) - found_instances))
+
+    assessment_list = []
+    for instance in instances:
+        assessment = {}
+        assessment['storage'] = {}
+        total_size_gb = 0
+        if instance.image:
+            image = glance.images.get(instance.image.get('id'))
+            image_size = math.ceil(image.size / (1024 ** 3))
+            total_size_gb += image_size
+            image_info = {"size_bytes": image.size,
+                          "source_image_name": image.name,
+                          "os_type": image.get("os_type", "linux")}
+            assessment["storage"]["image"] = image_info
+
+        volume_ids = [vol.id for vol in
+                      nova.volumes.get_server_volumes(instance.id)]
+        volumes = [cinder.volumes.find(id=vol_id) for vol_id in volume_ids]
+        total_size_gb += sum([volume.size for volume in volumes])
+        volumes_info = [{"volume_name": volume.name,
+                         "volume_id": volume.id,
+                         "size_bytes": 2**30 * volume.size}
+                        for volume in volumes]
+        assessment["storage"]["volumes"] = volumes_info
+
+        instance_flavor = nova.flavors.get(instance.flavor['id'])
+        total_size_gb += instance_flavor.disk
+        flavor_info = {"flavor_name": instance_flavor.name,
+                       "flavor_id": instance_flavor.id,
+                       "flavor_vcpus": instance_flavor.vcpus,
+                       "flavor_disk_size":
+                       instance_flavor.disk * (2 ** 30)}
+        assessment["storage"]["flavor"] = flavor_info
+
+        tenant_id = instance.tenant_id
+        tenant_name = source_client.get_project_name(tenant_id)
+        assessment["instance_name"] = instance.name
+        assessment["instance_id"] = instance.id
+        assessment["source_tenant_id"] = tenant_id
+        assessment["source_tenant_name"] = tenant_name
+        assessment["storage"]["total_size_gb"] = total_size_gb
+
+        assessment_list.append(assessment)
+
+    return assessment_list
+
+
+def migration_assess(source_client, coriolis, migration_id):
+
+    migration = coriolis.migrations.get(migration_id)
+
+    creation_timestamp = migration.tasks[0].updated_at
+    finish_timestamp = migration.tasks[-1].updated_at
+    creation_date = datetime.datetime.strptime(
+        creation_timestamp, '%Y-%m-%dT%H:%M:%S.%f')
+    finish_date = datetime.datetime.strptime(
+        finish_timestamp, '%Y-%m-%dT%H:%M:%S.%f')
+
+    interval_date = finish_date - creation_date
+
+    migration_instances = migration.instances
+    assessment_list = instances_assess(source_client, migration_instances)
+    for assessment in assessment_list:
+        assessment["migration"] = {}
+        assessment["migration"]["migration_id"] = migration.id
+        assessment["migration"]["migration_status"] = migration.status
+        assessment["migration"]["migration_time"] = str(interval_date)
```

Comment:
how does the time diff look when cast to a string?

Reply:
H:M:S, thankfully.